core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -837,6 +833,7 @@ public Integer getMappedPort(final int originalPort) {
         Preconditions.checkState(containerId != null, "Mapped port can only be obtained after the container is started");
```

Comment:
shouldn't `getMappedPort` be a default implementation in `ContainerState` interface? Ofc implementations like `DockerComposeContainer` will override it because of their specifics, but the rest should be fine with this implementation

Reply:
The `execInContainer()` methods, which have now been moved to `ExecInContainerPattern` still use a logger which must be passed into the method.

It's used for a debug and two trace statements, which are useful, but not exactly crucial.
Should they be removed, or can you see another way around this?